### ISSUE TITLE
HuBERT: Self-Supervised Speech Representation Learning by Masked Prediction of Hidden Units

### ISSUE THREAD

**Issue by ShotaroKataoka** [ShotaroKataoka/paper_summary#268]
## 0. 論文
タイトル：[HuBERT: Self-Supervised Speech Representation Learning by Masked Prediction of Hidden Units](https://arxiv.org/abs/2106.07447)  
著者：  [‪Wei-Ning Hsu‬](https://scholar.google.com/citations?user=N5HDmqoAAAAJ), Benjamin Bolte, Yao-Hung Hubert Tsai, Kushal Lakhotia, Ruslan Salakhutdinov, Abdelrahman Mohamed

arXiv投稿日：  14 Jun 2021
学会/ジャーナル：  [IEEE/ACM Transactions on Audio, Speech, and Language Processing](https://ieeexplore.ieee.org/xpl/RecentIssue.jsp?punumber=6570655)

[Wei-Ning Hsu](https://wnhsu.github.io/)さんは2023年現在はFacebook AI Researchにいるらしい？

他の論文
- [Audio-Visual HuBERT](https://ai.facebook.com/blog/ai-that-understands-speech-by-looking-as-well-as-hearing/): the first self-supervised model for audio-visual speech, achieving state-of-the-art performance on lip-reading, speech recognition, and audio-visual speech recognition with much less labeled data
- [data2vec](https://ai.facebook.com/blog/the-first-high-performance-self-supervised-algorithm-that-works-for-speech-vision-and-text/): The first high-performance self-supervised algorithm that works for speech, vision, and text
- [Textless Speech-to-Speech Translation on Real Data](https://arxiv.org/abs/2112.08352): first ever text-free speech-to-speech translation model trained on real data that is on par with text-based models
- [wav2vec-U](https://ai.facebook.com/blog/wav2vec-unsupervised-speech-recognition-without-supervision/): an unsupervised speech recognition framework that rivals the best supervised model from 2 years ago and works for 10 languages
- [Textless NLP](https://ai.facebook.com/blog/textless-nlp-generating-expressive-speech-from-raw-audio/): a model that can do prompted or unprompted speech generation without using any text (like audio-version of [GPT-2](https://openai.com/blog/better-language-models/))

## 1. どんなもの？
教師なしでスピーチ表現の自己教師学習を行う手法の提案。
k-meansによるシンプルな教師作成とCNNによる生の波形のインプットで学習し、SOTAであるwav2vecに匹敵する性能を示した。

## 2. 先行研究と比べてどこがすごい？
自己教師付き音声表現学習は近年注目されており、研究されている。
特にDiscreteBERTとこの研究は近い。
両方ともマスクされた部分の離散ターゲットを予測している。

DiscreteBERTとの違い：
- HuBERTは入力として生の波形データを受け取ることで、Transformerにより多くの情報を渡している（DiscreteBERTはたぶん離散化された入力を渡している。）
- DiscreteBERTはvq-wav2vec学習済みのユニットを使っているが、HuBERTはシンプルなk-meansターゲットを使いよりよい性能を達成した
- DiscreteBERTは教師が固定だったが、本研究では教師の質を向上させるためのテクニックを紹介する

wav2vec 2.0との違い：
- wav2vec 2.0では負のフレームをどこかからサンプリングする必要があるが、これを慎重に設定しないといけない。
- ほかにもあったけどめんどくさくなったので省略

## 3. 技術や手法のキモはどこ？
![image](https://user-images.githubusercontent.com/42331656/231319667-d7772302-1b39-4c8f-8265-92388e207b92.png)
- k-meansとかのクラスタリングのアンサンブルで教師を作成
- 生の波形をCNNでエンコードしてTransformerに入力

## 4. どうやって有効だと検証した？
![image](https://user-images.githubusercontent.com/42331656/231320026-4c34836f-49ce-4f3a-bd2e-15885d913f6c.png)
- Data
960 hours of `LibriSpeech audio` or 60,000 hours of `Libri-light audio` を事前学習に用いた。
- Unsupervised Unit Discovery
run k-means clustering with 100 clusters on 39-dimensional MFCC features
- Pre-Training
BASE model for two iterations on the 960 hours of LibriSpeech audio on 32 GPUs, with a batch size of at most 87.5 seconds of audio per GPU
Training for 100k steps takes about 9.5 hours. (250k steps / 1 iter)
- Metrics of Target Quality
phone purity (Phn Pur.)
cluster purity (Cls Pur.)
phone-normalized mutual information (PNMI)

Automatic speech recognition（ASR）で評価をした（たぶん）

## 5. 議論はある？
今後の課題
- 1フェーズで構成することを計画
- ASR以外の複数の下流タスクへの応用

## 6. 次に読むべき論文は？
- DiscreteBERT
- wav2vec 2.0
- この著者の他の研究

## 読中メモ
スピーチ表現の自己教師学習には３つのユニークな問題がある
１．それぞれの発話インプットに複数の音単位（sound units）が存在する
２．事前学習時に音単位の入力の辞書がない
３．音単位は様々な長さがあり明確な区切りがない
これらに対処するためにHidden-Unit BERT (HuBERT)を提案した。
これは自己教師スピーチ表現学習でありこれはオフラインのクラスタリングステップを用いて、BERTのような損失のためのアラインされたターゲットラベルを提供する。

このアプローチの重要な点：
マスクされた領域にのみ損失を適用する
→連続入力に対して音響と言語モデルを組み合わせた学習をモデルに強制すること。

教師なしクラスタリングのステップの一貫性に依存している。

まず、シンプルに100クラスタのk-meansから始め、そして2イテレーションのクラスタリングを行った。
すると、
```
state-ofthe-art wav2vec 2.0 performance on the Librispeech (960h) and Libri-light (60,000h) benchmarks with 10min, 1h, 10h, 100h, and 960h fine-tuning subsets

SOTAであるwav2vec 2.0の性能に匹敵した。
```

1Bパラメータを用いるとより困難なdev-otherおよびtest-other評価サブセットにおいて、最大19%および13%の相対WER削減を示した。

### method

- 半教師付き学習：テキストと音声のペアで学習したモデルを用いて強制アラインメントを行い、疑似的な音声ラベルをつくる。
- 自己教師付き表現学習：音声のみを用いる。何らかのアラインメントが必要。

これまでのアラインメント
- k-mensやガウス混合モデル(GMM)のような単純な離散潜在変数モデル
- より優れたグラフィカルモデル、ニューラルネットワークによるacoustic unit discovery models(音響ユニット発見モデル？)

音響ユニット発見モデルを用いてフレームレベルのターゲットを提供することを提案。
```
speech utterance: X = [x1, · · · , xT ] of T frames
Discovered hidden units: h(X) = Z = [z1, · · · , zT ], where zt ∈ [C] is a C-class 
```

たぶん：
入力波形とターゲットを考える。
入力波形はCNNでフレームごとにベクトル化したあと、Transformerに入力される。
ターゲットは入力波形をフレームごとにAcoustic Unit Discovery System(k-meansとかMFCC)によってクラスタリングしたもの。
Transformerはマスクされたフレームのベクトルがどのクラスタに属するかを予測する。

複数のクラスタリング手法をアンサンブルすることでターゲットの質を向上させる。

1時間15分
